chef/cookbooks/crowbar/templates/default/upgrade_cloud_node.sh.erb

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+#
+# This script starts the upgrade of the node server to the latest version of
+# system and Cloud product.
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+mkdir -p "`dirname "$LOGFILE"`"
+
+# Copy stdout to fd 3
+exec 3>&1
+# Create fd 4 for logfile
+exec 4>> "$LOGFILE"
+# Set fd 1 and 2 to logfile
+exec 1>&4 2>&1
+
+set -x
+
+initiate_node_upgrade()
+{
+    local installdir=/var/lib/crowbar/install
+    mkdir -p $installdir
+
+    if [[ -f $installdir/node-upgrading ]] ; then
+        echo "Exit: Upgrade is already running..."
+        exit 1
+    fi
+
+    # Signalize that the upgrade is running
+    touch $installdir/node-upgrading
+
+    # Upgrade the distribution non-interactively
+    zypper --no-color --releasever <%= @target_platform_version %> ref -f
+    zypper --no-color --non-interactive dist-upgrade -l --recommends --replacefiles
+    ret=$?
+    if [ $ret != 0 ]; then
+        echo "zypper dist-upgrade has failed with $ret, check zypper logs"
+        echo "$ret" > $installdir/node-upgrade-failed
+        # cleanup upgrading indication
+        rm -rf $installdir/node-upgrading
+        return
+    fi
+
+    # Signalize that the upgrade correctly ended
+    touch $installdir/node-upgraded-ok
```

Comment:
I just realized:

- we should probably not do anything from this script if that file exists
- we should somehow put a version with this file (either in the content, or in the filename) so that we know it's for the 6->7 upgrade